### PR TITLE
Allow configuring serverExternals

### PR DESCRIPTION
**Pull request by yavorsky**
### 🔦 Summary
This should help with:
https://jira.wixpress.com/projects/FEDINF/issues/FEDINF-1094

Like any other externals (webworkerExternals/externals) we will let the user configure it because they are changing according to the app/platform.

Thunderbolt's server (which downloads thunderbolt-elements) needs to verify those externals are located in the node_modules directory.

We are currently not supporting different types of externals in array, but it's pretty ok for start and not sure a lot of user's using it like this.

Will open a follow-up PRs to `thunderbolt-elements`, `editor-elements-library` and `thunderbolt-app`

### CODE REVIEW

File: packages/yoshi-common/src/webpack.config.ts [hunk 1]
```diff
@@ -1091,7 +1115,29 @@ export function createBaseWebpackConfig({
               // Svelte and React should always be external on the server side. Only relevant when more than one
```
Comment (ranyitz) [inline packages/yoshi-common/src/webpack.config.ts:1115]: This comment belongs to the "forceSpecificNodeExternals" line.

I would add a comment that explains that we let the user to configure the externals for the server.

File: packages/yoshi-flow-monorepo/src/webpack.config.ts [hunk 2]
```diff
@@ -200,6 +200,8 @@ export function createServerWebpackConfig(
     nodeExternalsWhitelist: libs.map(pkg => new RegExp(pkg.name)),
     useAssetRelocator: pkg.config.experimentalUseAssetRelocator,
     forceMinimizeServer: isThunderboltElementModule(pkg),
+    serverExternals: pkg.config.serverExternals,
+    // Remove next line when serverExternals will be implemented on thunderbolt app and element modules
```
Comment (ranyitz) [inline packages/yoshi-flow-monorepo/src/webpack.config.ts:204]: Actually, `thunderbolt-app` is a regular server that doesn't need externals, you probably mean `thunderbolt-elements`.

File: packages/yoshi-flow-monorepo/src/webpack.config.ts [hunk 3]
```diff
@@ -333,7 +331,7 @@ export function createSiteAssetsWebpackConfig(
     cwd: pkg.location,
     configName: 'site-assets',
     target: 'node',
-    useNodeExternals: false,
+    serverExternals: {},
```
Comment (ranyitz) [inline packages/yoshi-flow-monorepo/src/webpack.config.ts:334]: It worth adding a comment here, that this means we don't have any externals (even for node bundle).

**Comment by NitayRabi** [comment]
If I understand correctly - what we need to make sure that `react` (svelte is no longer relevant) should **NOT** be bundled into the server bundle, and that's supposed to be handled in package json configuration?

**Comment by yavorsky** [comment]
@NitayRabi yeah, in `yoshi.config.js` 🙂
Like here: https://github.com/wix-private/editor-elements/pull/614

**Comment by NitayRabi** [comment]
LGTM (from editor-elements point of view)

**Pull request merged**